### DISCUSSION TITLE
Separate PR per package

### DISCUSSION THREAD

**Discussion by maartenflippo** [release-plz/release-plz#1813] [Q&A] [answered]
Let me start by saying that this is a fantastic tool! Thank you for the effort you put into it.

We are coming from a workflow with `release-please`, in which we depend on its feature to open separate PRs per package in a cargo workspace. Ideally, we want to move this workspace to release-plz, but releasing packages individually is critical for now. Looking through the docs, however, it seems to me that it is not (yet) possible with release-plz. Is that true? I could not find any issues/PRs (closed or open) on the subject, but that may be poor querying from my side.

I did find this (source https://release-plz.ieni.dev/docs/github/output):

> prs: The release PRs opened by release-plz. It's an array of objects with the properties of pr. (Not useful for now. Use pr instead).

This makes me think it is a future feature, but it is not yet implemented.

**Comment by marcoieni** [comment]
Yes, it's a future feature. I would like to have this!

How does this feature work in release-please?

This is the closest issue open: https://github.com/MarcoIeni/release-plz/issues/1029

**Reply by maartenflippo** [reply]
Ah, so I did miss an issue. It seems a release group of size 1 would be what I am looking for.

In terms of how this works in release-please, instead of creating 1 release branch & pr, the branch and PR encode the package + new version that is to be released. Although I am not sure how release-please works specifically, I can imagine that this can be determined similarly to how the changelog is updated for separate packages. I.e., there it is necessary to determine which commits in the history apply to which package. If a package has at least one commit that could be a release, then create a branch&pr for that subset of changes. In terms of how this is done on an implementation side, I have no idea. But I can spend some time to figure that out if that would be helpful.

In general, do you currently think that release groups and this feature of one PR per package are the same feature? I would assume that the release groups feature is more complex, as the specific groups need to be determined. In case you think of the "1 per per package" feature as separate, I would be willing to spend some time to attempt a PR to implement this for release-plz.

**Reply by marcoieni** [reply]
> it is necessary to determine which commits in the history apply to which package. If a package has at least one commit that could be a release, then create a branch&pr for that subset of changes

We already have this info in release-plz because we update the changelog of every package individually, so we need to know the new commits for every package 👍

> do you currently think that release groups and this feature of one PR per package are the same feature?

I don't think so. We can separate for now and then rethink the config file. The important thing is to add this logic in release-plz 👍

I'm fine adding a separate config flag called `one_pr_per_package` or a clearer name.

> I would be willing to spend some time to attempt a PR to implement this for release-plz.

Go for it! Please create an issue first, so I can assign it to you 👍

**Reply by maartenflippo** [reply, marked as answer]
I opened issue https://github.com/MarcoIeni/release-plz/issues/1814
